vpdq/cpp/hashing/matchTwoHash.cpp

```diff
@@ -2,82 +2,99 @@
-  vector<vpdq::hashing::vpdqFeature> queryFiltered, targetFiltered;
-  size_t i = 0, j = 0;
-  int qMatchCnt = 0;
-  int tMatchCnt = 0;
-  for (i = 0; i < qHashes.size(); i++) {
-    if (qHashes[i].quality >= qualityTolerance) {
-      queryFiltered.push_back(qHashes[i]);
-    } else if (verbose) {
-      printf(
-          "Skipping Line %zu Skipping Query Hash: %s, because of low quality Query Hash: %d \n",
-          i,
-          qHashes[i].pdqHash.format().c_str(),
-          qHashes[i].quality);
-    }
-  }
-  for (i = 0; i < tHashes.size(); i++) {
-    if (tHashes[i].quality >= qualityTolerance) {
-      targetFiltered.push_back(tHashes[i]);
+
+/**
+ * @brief Filter low quality hashes from a feature vector
+ *
+ * @param features Features to filter
+ * @param qualityTolerance Quality tolerance of comparing two hashes. If lower
+ * then it won't be included in the result
+ * @param verbose Print skipped hashes
+ *
+ * @return Feature vector without features with quality lower than
+ * qualityTolerance
+ */
+static std::vector<vpdq::hashing::vpdqFeature> filterFeatures(
+    const std::vector<vpdq::hashing::vpdqFeature>& features,
+    const int qualityTolerance,
+    const bool verbose) {
+  std::vector<vpdq::hashing::vpdqFeature> filteredHashes;
+  for (const auto& feature : features) {
+    if (feature.quality >= qualityTolerance) {
+      filteredHashes.push_back(feature);
     } else if (verbose) {
-      printf(
-          "Skipping Line %zu Skipping Target Hash: %s, because of low quality Target Hash: %d \n",
-          i,
-          tHashes[i].pdqHash.format().c_str(),
-          tHashes[i].quality);
+      auto index = &feature - &features[0];
+      std::cout << "Skipping Line " << index
+                << " Skipping Hash: " << feature.pdqHash.format()
+                << ", because of low quality: " << feature.quality << std::endl;
     }
   }
+  return filteredHashes;
+}
 
-  for (i = 0; i < queryFiltered.size(); i++) {
-    for (j = 0; j < targetFiltered.size(); j++) {
-      if (queryFiltered[i].pdqHash.hammingDistance(targetFiltered[j].pdqHash) <
-          distanceTolerance) {
-        qMatchCnt++;
-        if (verbose) {
-          printf(
-              "Query Hash: %s Target Hash: %s match \n",
-              queryFiltered[i].pdqHash.format().c_str(),
-              targetFiltered[j].pdqHash.format().c_str());
-        }
-        break;
-      }
-    }
-  }
-  for (i = 0; i < targetFiltered.size(); i++) {
-    for (j = 0; j < queryFiltered.size(); j++) {
-      if (targetFiltered[i].pdqHash.hammingDistance(queryFiltered[j].pdqHash) <
+/**
+ * @brief Get the number of matches between two feature vectors
+ *
+ * @param features1 Features to match
+ * @param features2 Features to match
+ * @param distanceTolerance Distance tolerance of considering a match. Lower is
+ * more similar.
+ * @param verbose Print features with matching hashes
+ *
+ * @return Number of matches
+ */
+static unsigned int findMatches(
```

Comment:
I last used C++ seriously in 2014, so I might be a yokel :P 

Rather than `unsigned int` does `size_type` align better, since it should be the same range as the number of elements in the vectors.

Reply:
Good catch. I'll change it.